### ISSUE TITLE
An error occurs if a project has multiple auto-deploy triggers

### ISSUE THREAD

**Issue by droyad** [OctopusDeploy/Issues#5482]
Introduced in `2019.3.1` (#5459) and `2019.4.0` (#5421)

If a project has multiple deployment target auto-deploy triggers, the following error can occur when the background auto-deploy process runs:

```
Error when running scheduled task: Octopus.Server.Schedules.RunOnAScheduleAdapter`1[Octopus.Server.Schedules.ProcessAutoDeployments] System.ArgumentException: An item with the same key has already been added.
   at System.ThrowHelper.ThrowArgumentException(ExceptionResource resource)
   at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
   at System.Linq.Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector, IEqualityComparer`1 comparer)
   at Octopus.Server.Schedules.AutoDeployments.MachineFilters.ProjectTenantedDeploymentModeMachineFilter..ctor(ILog log, IReadOnlyList`1 projects)
   at Octopus.Server.Schedules.AutoDeployments.AutoDeployMachineFiltererFactory.Create(IReadOnlyList`1 projects)
   at Octopus.Server.Schedules.AutoDeployments.AutoDeployItemBuilder.Build(IReadOnlyList`1 autoDeployTriggers, IReadOnlyList`1 latestSuccessfulDeployments, IReadOnlyList`1 projects, IReadOnlyList`1 autoDeployMachines)
```

# Steps to reproduce

1. Add more than one deployment target auto-deploy trigger to a project.
2. Disable and enable a deployment target which causes the auto-deploy process to do something.
3. Look in the Octopus Server logs where this error will appear.

# Investigation

This is due to to duplicate projects being loaded here https://github.com/OctopusDeploy/OctopusDeploy/blob/002df1007071e5eec44166046592f6ba5f85ba90/source/Octopus.Server/Schedules/AutoDeployments/AutoDeployTriggerProcessor.cs#L96

# Versions

- **Octopus Server** - Introduced in `2019.3.1` (#5459) and `2019.4.0` (#5421)
Fixed in `2019.3.3` and `2019.4.7`.

# Workaround

A work around is to disable all except one of the triggers.

# Links

https://help.octopus.com/t/auto-deploy-trigger-issues/22714

**Comment by SergeyLukashov** [comment]
Hey @michaelnoonan Just wonder is there any updates on this issue? This thing is really affecting our prod system - we are currently running dozen of instances 24/7 instead of doing some intelligent auto-scaling.

**Comment by michaelnoonan** [comment]
Hi @SergeyLukashov I'm planning to ship this fix in `2019.3.3` on Thursday this week AEST. Hope that helps.

**Comment by michaelnoonan** [comment]
Release Note: Fixed a bug preventing deployment target triggers from working when any project has more than one deployment target trigger defined

**Comment by michaelnoonan** [comment]
This shipped in 2019.3.3 LTS.
